frontend/src/Components/SpinnerIcon.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
```

Comment:
Did the same refactor on another branch
Here's my version

```suggestion
import React from 'react';
import { icons } from 'Helpers/Props';
import Icon, { IconProps } from './Icon';

export interface SpinnerIconProps extends IconProps {
  spinningName?: IconProps['name'];
  isSpinning: Required<IconProps['isSpinning']>;
}

export default function SpinnerIcon({
  name,
  spinningName = icons.SPINNER,
  ...otherProps
}: SpinnerIconProps) {
  return (
    <Icon
      name={(otherProps.isSpinning && spinningName) || name}
      {...otherProps}
    />
  );
}
```

Reply:
Totally spaced that you did this one already, making a small tweak to not use `otherProps` inside of the component at the cost of a couple extra lines 😄